File: frontend/src/api/admin/searchProvider.ts
import apiClient from '../client'

export interface SearchProvider {
  id: string
  name: string
  provider_type: 'api' | 'searxng' | 'scrape'
  api_key: string
  base_url: string
  is_enabled: boolean
  is_default: boolean
  params: { maxResults?: number; searchDepth?: string }
  timeout_seconds: number
  key_url: string
  is_local: boolean
}

export const searchProviderApi = {
  async list(): Promise<SearchProvider[]> {
    const { data } = await apiClient.get('/admin/search-providers')
    return data || []
  },
  async update(id: string, payload: Partial<SearchProvider>): Promise<SearchProvider> {
    const { data } = await apiClient.put(`/admin/search-providers/${id}`, payload)
    return data
  },
  async setDefault(id: string): Promise<SearchProvider> {
    const { data } = await apiClient.put(`/admin/search-providers/${id}/default`)
    return data
  },
  async test(id: string): Promise<{ success: boolean; message: string; result_count: number }> {
    const { data } = await apiClient.post(`/admin/search-providers/${id}/test`)
    return data
  },
}

File: src/omichub/application/schemas/search_provider.py
"""联网搜索服务商 DTO。"""

from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class SearchProviderDTO(BaseModel):
    id: str
    name: str
    provider_type: str
    api_key: str = ""
    base_url: str = ""
    is_enabled: bool = False
    is_default: bool = False
    params: dict[str, Any] = Field(default_factory=dict)
    timeout_seconds: int = 10
    key_url: str = ""
    is_local: bool = False
    updated_at: datetime | None = None


class SearchProviderUpdateDTO(BaseModel):
    api_key: str = ""
    base_url: str = ""
    is_enabled: bool = False
    params: dict[str, Any] = Field(default_factory=dict)
    timeout_seconds: int = Field(default=10, ge=1, le=10)


class SearchProviderTestResponse(BaseModel):
    success: bool
    message: str = ""
    result_count: int = 0


class SearchProviderAvailabilityDTO(BaseModel):
    available: bool
    provider_name: str = ""
